### ISSUE TITLE
gravity.sh invalid operator syntax error (plural detection)

### ISSUE THREAD

**Issue by dracenmarx** [pi-hole/pi-hole#1808]
**In raising this issue, I confirm the following:** `{please fill the checkboxes, e.g: [X]}`

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [x] The issue I am reporting can be *replicated*.
- [x] The issue I am reporting isn't a duplicate (see [FAQs](https://github.com/pi-hole/pi-hole/wiki/FAQs), [closed issues](https://github.com/pi-hole/pi-hole/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20), and [open issues](https://github.com/pi-hole/pi-hole/issues)).

**How familiar are you with the the source code relevant to this issue?:**

4

---
**Actual behavior**

/opt/pihole/gravity.sh shows following error (German):

> /opt/pihole/gravity.sh: Zeile 455: [[: 44.559: Syntaxfehler: Ungültiger arithmetischer Operator. (Fehlerverursachendes Zeichen ist \".559\").

Line 455 has following content:
`plural=""; [[ "${num}" -ne 1 ]] && plural="s"`

The problem is that $num has thousand separators, which cannot be parsed by "-ne".
Actually, $num is "44.559"  (fullstop is the German thousand separator)

**Fix:**

Find (2 times):
`plural=; [[ "${num}" -ne 1 ]] && plural="s"`

Replace (2 times):
`plural=; [ "${num}" != "1" ] && plural="s"`

**Comment by Mcat12** [comment]
If you want to open a PR, we'll review it and merge. Or we will get around to it soon.

**Comment by dracenmarx** [comment]
I don't know how to create a pull-request, so it would be great if you could apply the change directly. :-)

**Comment by Mcat12** [comment]
@dracenmarx Please test the changes from the above pull request.

**Comment by dracenmarx** [comment]
@Mcat12 I have found some problems and added them in the review.
**But please don't submit this pull request**. I have found a better solution.

With the current version, 2 of 5 displayed values don't get thousand separators.
To have thousand separators for all 5 values and make everything more simple (e.g. allow the calculation `num=$(( num/2 )`, I suggest that the thousand-separator-addition is applied directly in the `echo` commands.

Here is the diff of my new change suggestion. I'm sorry I didn't figure out how to do a pull request yet.

```
[11:26]root@ras3:/tmp$ diff -U 0 original patch
--- original    2017-12-10 11:22:41.978377259 +0100
+++ patch       2017-12-10 11:25:41.268592074 +0100
@@ -394 +394 @@
-  num=$(printf "%'.0f" "$(wc -l < "${piholeDir}/${parsedMatter}")")
+  num=$(wc -l < "${piholeDir}/${parsedMatter}")
@@ -396 +396 @@
-  ${INFO} ${COL_BLUE}${num}${COL_NC} domains being pulled in by gravity"
+  ${INFO} ${COL_BLUE}$(printf "%'.0f" "$num")${COL_NC} domains being pulled in by gravity"
@@ -404,2 +404,2 @@
-  num=$(printf "%'.0f" "$(wc -l < "${piholeDir}/${preEventHorizon}")")
-  echo -e "  ${INFO} ${COL_BLUE}${num}${COL_NC} unique domains trapped in the Event Horizon"
+  num=$(wc -l < "${piholeDir}/${preEventHorizon}")
+  echo -e "  ${INFO} ${COL_BLUE}$(printf "%'.0f" "$num")${COL_NC} unique domains trapped in the Event Horizon"
@@ -438 +438 @@
-  str="Whitelisting ${num} domain${plural}"
+  str="Whitelisting $(printf "%'.0f" "$num") domain${plural}"
@@ -452 +452 @@
-    num=$(printf "%'.0f" "$(wc -l < "${blacklistFile}")")
+    num=$(wc -l < "${blacklistFile}")
@@ -454 +454 @@
-    echo -e "  ${INFO} Blacklisted ${num} domain${plural}"
+    echo -e "  ${INFO} Blacklisted $(printf "%'.0f" "$num") domain${plural}"
@@ -464 +464 @@
-    echo -e "  ${INFO} Wildcard blocked ${num} domain${plural}"
+    echo -e "  ${INFO} Wildcard blocked $(printf "%'.0f" "$num") domain${plural}"
```

**Comment by jacobsalmela** [comment]
Please look at #1850 for this issue

**Comment by jacobsalmela** [comment]
#1850 will take care of this issue.

**Comment by dschaper** [comment]
Closed via #1856
